### DISCUSSION TITLE
How to specify build and run command?

### DISCUSSION THREAD

**Discussion by ghost** [orbitalquark/textadept#360] [Q&A]
For example, the language is `Foo`, the source file extension is `.foo`. The compiler (to bytecode) is `foo-compiler` and it will compile `test.foo` to `test.run` (executable) or `test.lib` (library). The vm is `foo-runner`, the executable can be run with `foo-runner test.run`. The debugger is `foo-dbg`.

**Comment by orbitalquark** [comment]
The FAQ points you to this: 
https://orbitalquark.github.io/textadept/manual.html#textadept. There 
are examples of setting up run commands there. The LuaDoc also covers 
other commands like compile and build: 
https://orbitalquark.github.io/textadept/api.html#textadept.run

Perhaps something like:

   textadept.run.compile_commands.Foo = 'foo-compiler "%f"'
   textadept.run.run_commands.Foo = 'foo-run "%f"'
